Add docs for validity middleware, validity middleware mock, and focus middleware mock (#669)
dojo/framework: docs/en/testing/supplemental.md

> src/Breakpoint.tsx

```tsx
import { tsx, create } from '@dojo/framework/core/vdom';
import breakpoint from '@dojo/framework/core/middleware/breakpoint';

const factory = create({ breakpoint });

export default factory(function Breakpoint({ middleware: { breakpoint } }) {

> 	const bp = breakpoint.get('root');

Reviewer: how come this was reformatted?

Author: @agubler This was the result of running prettier against the markdown files. I believe that example was using spaces instead of tabs and Prettier corrected it.

Reviewer: Oh weird running prettier against master doesn't do anything for me.

});
```

#### Mock `focus` middleware

> Using `createFocusMock` from `@dojo/framework/testing/middleware/focus` offers tests manual control over when the `focus` middleware reports that a node with a specified key is focused.

Reviewer 2: ```suggestion
Using `createFocusMock` from `@dojo/framework/testing/middleware/focus` provides tests with manual control over when the `focus` middleware reports that a node with a specified key gets focused.
```

Consider the following widget:

> src/FormWidget.tsx

```tsx
import { tsx, create } from '@dojo/framework/core/vdom';
import focus, { FocusProperties } from '@dojo/framework/core/middleware/focus';
import * as css from './FormWidget.m.css';

export interface FormWidgetProperties extends FocusProperties {}

const factory = create({ focus }).properties<FormWidgetProperties>();

export const FormWidget = factory(function FormWidget({ middleware: { focus } }) {
	return (
		<div key="wrapper" classes={[css.root, focus.isFocused('text') ? css.focused : null]}>
			<input type="text" key="text" value="focus me" />
		</div>
	);
});
```

> By calling `focusMock(key: string | number, value: boolean)` the result of the focus middleware's `isFocused` method can be controlled during a test.

Reviewer 2: ```suggestion
By calling `focusMock(key: string | number, value: boolean)` the result of the focus middleware's `isFocused` method can get controlled during a test.
```

});
```

#### Mock `validity` middleware

> Using `createValidityMock` from `@dojo/framework/testing/mocks/middleware/validity` creates a mock validity middleware where the return value of the `get` method can be controlled in a test.

Reviewer 2: ```suggestion
Using `createValidityMock` from `@dojo/framework/testing/mocks/middleware/validity` creates a mock validity middleware where the return value of the `get` method can get controlled in a test.
```

Consider the following example:

> src/FormWidget.tsx

```tsx
import { tsx, create } from '@dojo/framework/core/vdom';
import validity from '@dojo/framework/core/middleware/validity';
import icache from '@dojo/framework/core/middleware/icache';
import * as css from './FormWidget.m.css';

const factory = create({ validity, icache });

export const FormWidget = factory(function FormWidget({ middleware: { validity, icache } }) {
	const value = icache.getOrSet('value', '');
	const { valid, message } = validity.get('input', value);

	return (
		<div key="root" classes={[css.root, valid === false ? css.invalid : null]}>
			<input type="email" key="input" value={value} onchange={(value) => icache.set('value', value)} />
			{message ? <p key="validityMessage">{message}</p> : null}
		</div>
	);
});
```

> Using `validityMock(key: string, value: { valid?: boolean, message?: string; })`, the results of the validity mock's `get` method can be controlled in a test.

Reviewer 2: ```suggestion
Using `validityMock(key: string, value: { valid?: boolean, message?: string; })`, the results of the validity mock's `get` method can get controlled in a test.
```

> tests/unit/FormWidget.tsx

```tsx
const { describe, it } = intern.getInterface('bdd');
import { tsx } from '@dojo/framework/core/vdom';
import harness from '@dojo/framework/testing/harness';
import validity from '@dojo/framework/core/middleware/validity';
import createValidityMock from '@dojo/framework/testing/mocks/middleware/validity';
import * as css from './FormWidget.m.css';

describe('Validity', () => {

> 	it('adds a "invalid" class to the wrapper when the input is invalid and displays a message', () => {

Reviewer 2: ```suggestion
	it('adds the "invalid" class to the wrapper when the input is invalid and displays a message', () => {
```
